File: 3461-check-if-digits-are-equal-in-string-after-operations-i.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        while len(s) > 2:
            temp = ''
            for i in range(len(s) - 1):
                temp += str((ord(s[i]) - ord('0') +
                            ord(s[i+1]) - ord('0')) % 10)

            s = temp

        return s[0] == s[1]


sol = Solution()
print(f'output: {sol.hasSameDigits('3902')}, expected: True')
print(f'output: {sol.hasSameDigits('34789')}, expected: False')

File: 3461-check-if-digits-are-equal-in-string-after-operations-i.java
class CheckIfDigitsAreEqualInStringAfterOperationsI {

    public boolean hasSameDigits(String s) {
        int n = s.length();
        int[] digits = new int[n];

        for (int i = 0; i < n; i++)
            digits[i] = s.charAt(i) - '0';
        while (n > 1) {
            for (int i = 0; i < s.length() - 1; i++) {
                digits[i] = (digits[i] + digits[i + 1]) % 10;
            }
            n--;
            if (n == 2 && digits[0] == digits[1])
                return true;
        }

        return false;
    }

    public static void main(String[] args) {
        CheckIfDigitsAreEqualInStringAfterOperationsI sol = new CheckIfDigitsAreEqualInStringAfterOperationsI();
        System.out.println("Output: " + sol.hasSameDigits("3902") + ", expected: true");
        System.out.println("Output: " + sol.hasSameDigits("34789") + ", expected: false");
    }
}
